### PR TITLE
Introducing CommandController strategy and base implementations.

### PR DESCRIPTION
**Pull request by schwadorf**
Introducing CommandController strategy and base implementations:
- MapBasedCommandController (deprecating NewCommandController)
- SpringContextBasedInteractionCommandController
- SpringContextAwareInteractionCommandController

Airline example updated to use new spring-based implementation of CommandController.

### CODE REVIEW

File: interaction-core/src/main/java/com/temenos/interaction/core/command/NewCommandController.java [hunk 1]
```diff
@@ -49,6 +56,7 @@ public NewCommandController() {
 	public NewCommandController(Map<String, InteractionCommand> commands) {
 		assert(commands != null);
 		for(String name : commands.keySet()) {
+		    logger.warn("PETE  --  Adding new command: " + name);
```
Comment (andrewmcguinness) [inline interaction-core/src/main/java/com/temenos/interaction/core/command/NewCommandController.java]: Can this be removed or moved to a lower logging level please

Comment (schwadorf) [inline interaction-core/src/main/java/com/temenos/interaction/core/command/NewCommandController.java]: Sorry about that, it was left from a dirty commit. Will fix now.

File: interaction-parent/pom.xml [hunk 2]
```diff
@@ -39,8 +39,8 @@
 		<module>../interaction-media-odata-xml</module>
 		<module>../interaction-dsl/interaction-springdsl</module>
 		<module>../interaction-dynamic-loader</module>	
-		<module>../interaction-commands-jdbc</module>
-		<module>../interaction-jdbc-producer</module>
+		<!--module>../interaction-commands-jdbc</module>
+		<module>../interaction-jdbc-producer</module-->
```
Comment (andrewmcguinness) [inline interaction-parent/pom.xml]: Is this intended? looks like a merge issue

File: interaction-core/src/main/java/com/temenos/interaction/core/command/SpringContextBasedInteractionCommandController.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+
+/**
+ * Implementation of Implementation of {@link CommandController} delegating the
```
Comment (andrewmcguinness) [inline interaction-core/src/main/java/com/temenos/interaction/core/command/SpringContextBasedInteractionCommandController.java]: Minor typo here

File: interaction-dynamic-loader/src/main/java/com/temenos/interaction/loader/detector/ResourceChangeDetector.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * 
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+import java.util.Collection;
+
+/**
+ * TODO: Document me!
```
Comment (andrewmcguinness) [inline interaction-dynamic-loader/src/main/java/com/temenos/interaction/loader/detector/ResourceChangeDetector.java]: yep

**Pull request merged**